python/cudf/cudf/core/frame.py

```diff
@@ -163,6 +163,16 @@ def _from_columns(
 
         return cls._from_data(data, index)
 
+    def _from_columns_like_self(
+        self,
+        columns: List[ColumnBase],
+        column_names: List[str],
+        index_names: Optional[List[str]] = None,
+    ):
```

Comment:
A small description would be better.

Reply:
Agreed, this will become common enough to need a docstring.